Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM ghcr.io/astral-sh/uv:0.6.9-python3.10-bookworm@sha256:f4f319fff1ed098c4a195cdecd372e78cb97d0066faf3318fba4edfd1c7d522e as builder
+ENV UV_COMPILE_BYTECODE=1 UV_LINK_MODE=copy
+ENV UV_PYTHON_DOWNLOADS=0
+
+# Change the working directory to the `app` directory
+WORKDIR /app
+
+# Install dependencies
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-install-project --no-dev
+
+# Copy the project into the intermediate image
+COPY . /app
```

Comment:
Instead of copying everything, it's safer to enumerate files/dirs that are really needed please, like in https://github.com/gnosis/prediction-market-agent/blob/main/Dockerfile#L29

Reply:
The problem with `.dockerignore` is that it's easy to forget, and no one will notice it -- that happened to me with the historical version of the .env file previously.

Reply:
That's fair. Implementing like that but still populating .dockerignore for extra safety.